PhysicsTools/PatAlgos/python/slimming/puppiForMET_cff.py

```diff
@@ -18,25 +18,26 @@ def makePuppies( process ):
 
 
     process.puppiNoLep = process.puppi.clone()
-    process.puppiNoLep.candName = cms.InputTag('pfNoLepPUPPI') 
-
-    process.puppiMerged = cms.EDProducer("CandViewMerger",src = cms.VInputTag( 'puppiNoLep','pfLeptonsPUPPET'))
+    process.puppiNoLep.candName = cms.InputTag('particleFlow')
+    process.puppiNoLep.useWeightsNoLep = cms.bool(True)
+    
+#    process.puppiMerged = cms.EDProducer("CandViewMerger",src = cms.VInputTag( 'puppiNoLep','pfLeptonsPUPPET'))
```

Comment:
@mariadalfonso 
did you refer to 
https://github.com/cms-sw/cmssw/pull/15562/files#diff-01ee5c30d9563d02ea8692a76e26f625
?
Due to conflicting edits, and apparent need for changes in #15562 to go in first, complete review of this PR will need to wait for #15562 to be merged.